bdd4j-junit5/src/main/java/org/bdd4j/ScenarioBuilder.java

```diff
@@ -0,0 +1,52 @@
+package org.bdd4j;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The {@link ScenarioBuilder} provides access to available test {@link Step Steps} which can in turn be used
+ * to set up the test scenario.
+ * A test scenario is set up by adding steps using the addSteps method.
+ */
+public final class ScenarioBuilder<T extends BDD4jSteps<?>> {
+  private final T availableSteps;
+  private List<Step<?>> registeredSteps = Collections.emptyList();
+
+  /**
+   * Creates a new instance.
+   *
+   * @param availableSteps The available steps.
+   */
+  public ScenarioBuilder(final T availableSteps) {
+    this.availableSteps = availableSteps;
+  }
+
+  /**
+   * Registers the given steps with the scenario.
+   *
+   * @param steps The steps that should be registered.
+   * @param <S>   The type of the steps.
+   */
+  @SafeVarargs
+  public final <S> void registerSteps(final Step<S>... steps) {
```

Comment:
I believe register is not the correct term here.
When you register something, it ends up in a registry and might be used or not.
The added steps on the other hand are added to a list that respects the order and "guarantees" that the steps are executed in that order during the test run.
Maybe "defineSteps" or "defined orderedSteps" is a more fitting name for that method.